From 767c26e4696401059325c5eba07ca8db67d75885 Mon Sep 17 00:00:00 2001
From: "Richard.Wang" <wangleihd@gmail.com>
Date: Fri, 4 Dec 2015 21:14:01 -0800
Subject: [PATCH 09/12] Exmaple 8: Add write operation for /dev/hello

User:
	$ sudo insmod hello.ko
	$ dmesg
	$ sudo lsmod
	$ sudo mknod /dev/hello c 245 0
	$ cat /dev/hello
	$ sudo chmod 777 /dev/hello
	$ echo "Welcome to kernel" > /dev/hello
	$ cat /dev/hello
	$ sudo rmmod hello
	$ dmesg

diff --git a/hello.c b/hello.c
index 0c8217f..176884a 100644
--- a/hello.c
+++ b/hello.c
@@ -34,10 +34,24 @@ static ssize_t hello_read(struct file *filp, char __user *buff, size_t count, lo
 	return ret;
 }
 
+static ssize_t hello_write(struct file *filp, const char __user * buff, size_t count, loff_t *offp){
+	ssize_t ret  = 0;
+	if(count > MAX){
+		ret = -EFAULT;
+	}else{
+		memset(data, 0, MAX);
+		copy_from_user(data, buff, count);
+		data[count] = '\0';
+		ret = count;
+	}
+	return count;
+}
+
 static struct file_operations fops={
 	.owner = THIS_MODULE,
 	.open  = hello_open,
 	.read  = hello_read,
+	.write = hello_write,
 };
 
 static void char_reg_cdev(void){
-- 
1.7.9.5

